packages/manager/apps/dedicated/client/app/ip/firewall-game/rule/add/add.routing.js

```diff
@@ -0,0 +1,29 @@
+import isEmpty from 'lodash/isEmpty';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('app.ip.firewall-game.rule.add', {
+    url: '/add',
+    params: {
+      ip: {
+        type: 'string',
+      },
+      ipBlock: {
+        type: 'string',
+      },
+    },
+    redirectTo: (transition) => {
+      return Promise.all([
```

Comment:
Hmm should not you use `$q` for the moment ?

Reply:
This is the way we do it all the time with `redirectTo` and we've never had an issue for now :)